packages/editor-ui/src/components/Modal.vue

```diff
@@ -126,60 +136,60 @@ export default Vue.extend({
 		},
 	},
 	computed: {
-		width(): string {
-			return this.$props.size ? sizeMap[this.$props.size] : sizeMap.default;
-		},
 		isActive(): boolean {
 			return this.$store.getters['ui/isModalActive'](this.$props.name);
 		},
-		dialogVisible(): boolean {
+		visible(): boolean {
 			return this.$store.getters['ui/isModalOpen'](this.$props.name);
 		},
+		styles() {
+			const styles: {[prop: string]: string} = {};
+			if (this.height) {
+				styles['--dialog-height'] = this.height;
+			}
+			if (this.maxHeight) {
+				styles['--dialog-max-height'] = this.maxHeight;
+			}
+			if (this.maxWidth) {
+				styles['--dialog-max-width'] = this.maxWidth;
+			}
+			if (this.minWidth) {
+				styles['--dialog-min-width'] = this.minWidth;
+			}
+			return styles;
+		},
 	},
 });
 </script>
 
 <style lang="scss">
-.el-drawer__header {
-	margin: 0;
-	padding: 30px 30px 0 30px;
-}
-
-.el-drawer__body {
-	overflow: hidden;
-}
-
-.el-dialog__body {
-	height: 100%;
-}
-
 .dialog-wrapper {
-	&.xl > div, &.md > div {
-		min-width: 620px;
+	.el-dialog {
+		display: flex;
+		flex-direction: column;
+		max-width: var(--dialog-max-width, 80%);
+		min-width: var(--dialog-min-width, 420px);
+		height: var(--dialog-height);
+		max-height: var(--dialog-max-height);
 	}
 
-	&.lg > div {
-		height: 80%;
+	.el-dialog__body {
 		overflow: hidden;
+		display: flex;
+		flex-direction: column;
+		flex-grow: 1;
+	}
 
-		.modal-content {
-			height: 100%;
-		}
+	.modal-content {
+		overflow: auto;
+		flex-grow: 1;
```

Comment:
allows for our dream setup for content to collapse while footer remains at bottom... now implemented for workflow settings